runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -753,8 +763,19 @@ private void pruneRemovedRouters(@Nonnull Layout layout) {
      */
     private CompletableFuture<Layout> fetchLayout(List<String> layoutServers) {
 
+        // Layout fetch can result in an exception if the systemDownHandler was invoked and the
+        // completable future was completed exceptionally.
+        Layout lastKnownLayout;
+        try {
+            lastKnownLayout = layout != null ? CFUtils.getUninterruptibly(layout) : null;
+        } catch (Throwable th) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)